### PR TITLE
Fixes for editing Authors and adds Typeahead for Author input

### PR DESCRIPTION
**Pull request by cervinko**
I had a lot of Authors 'Unbekannt' in my library. When i entered a Author in the Web interface all 'Unbekannt' where edited. With this fix a new Author is added to the db and linked to the book.
It would be better to use gettext instead of `if book.authors[0].name not in  ("Unknown", "Unbekannt", "", " "):` but that seemed to complex for me.

Added Typeahead for Author input

thats my first github pull request. i wanted to seperate it in two parts; didn't work.

### CODE REVIEW

**Comment by janeczku** [comment]
Great PR, thanks! Will have a look at it soon!

**Comment by janeczku** [comment]
Just had a quick look. Two problems came up:

1. Given the current author name is "Rolf Schmidt" and i correct the spelling to "Rolf Schmid" then the change is not  applied.
2. Given the current author name is "John Doe" and i correct the name to "Jane Doe"  then "John Doe" remains in the authors table even though there is no book associated with it.

**Comment by cervinko** [comment]
Thank you for the feedback. Unfortunatelly i can not reproduce your the bugs. Did you look up the metadatachanges in the Authors List page? Because this one uses the 'sort' column of the authors table and this column is not edited by calibre-web at all. (but it clearly should be one day)

Otherwise 'Rolf Schmidt' should be changed to 'Rolf Schmid' for all books that had 'Rolf Schmidt' as author. The same with 'John Doe'. It changes the db entry of 'John Doe' but does not add a new entry. A new author entry is only added when the previous Authorname was 'Unbekannt', 'Unknown, ' '  or ''.

File: cps/web.py [hunk 1]
```diff
@@ -655,7 +664,25 @@ def edit_book(book_id):
     if request.method == 'POST':
         to_save = request.form.to_dict()
         book.title = to_save["book_title"]
-        book.authors[0].name = to_save["author_name"]
+        
+        author_id = book.authors[0].id
+        
+        is_author = db.session.query(db.Authors).filter(db.Authors.name.like('%' + to_save["author_name"].strip() + '%')).first()
```
Comment (janeczku) [inline cps/web.py]: Why are you doing a wildcard match of the author name?
The wildcard will match to_save["author_name"]="Rolf Schmid" with an existing author "Rolf Schmidt". Thats why the book author doesn't get changed in that case as i previously commented.
It should be an equal operator here.

Comment (cervinko) [inline cps/web.py]: thx!. changed it to equal operator

**Comment by janeczku** [comment]
Nice. Thanks a lot for your contribution!

**Pull request merged**